CHANGELOG.md

```diff
@@ -1,3 +1,73 @@
+## 2023-05-16 v0.17.0
+
+## What's Changed
+
+Release highlights:
+* new rest api for Waku Store protocol has been added
+* Waku2 Swap protocol has been removed
+* new Filter protocol implentations has been incorporated
+* initial C bindings support has been merged
+* support for Heaptrack to investigate memory utilization has been added
+* the `topics` parameter has been deprecated in wakunode2 app in favour of the `topic` parameter
```

Comment:
I'd not include this in the highlights, but just a single line under a new section `Upgrade instructions` as in https://github.com/waku-org/nwaku/releases/tag/v0.15.0, which states:

> Note that the `--topics` config item has been deprecated and support will be dropped in future releases. To configure support for multiple pubsub topics, use the new `--topic` parameter repeatedly.

I usually similarly add a > Note under the release title to bring people's attention to the fact that this release comes with Upgrade Instructions.